### DISCUSSION TITLE
Permeability of excavated tunnel - convergence problem

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#27414] [Q&A Modules: Porous Flow]
Hi PorousFlow experts,

I checked https://mooseframework.inl.gov/modules/porous_flow/coal_mining.html I noticed that the excavted_perm is set to 0

I find that having a very low perm at the excavated elements make the simulation very difficult to converge.

So here is my setup, I have activate drain BC around the tunnel (below)

```
  [bc_drain0001_UG]
    type = PorousFlowPiecewiseLinearSink
    variable = porepressure_L1
    boundary = 'DRAINSRFUPIT_YXXXX_QXX'
    pt_vals = '0 1e9'
    multipliers = '0 1e9'
    flux_function = 1
    PT_shift = 0.
    fluid_phase = 0
    use_mobility = true    # default is false
  []
```

on top of that, the flow model is partially saturated with + FLAC relperm m = 2, so this make the perm in the excavated area even lower (which doesn't really help with convergence), I set the sat_lr = 0.25 and it still doesnt really help. in fact I'd like to set alpha to be higher than 1e-6 but the convergence will be much worse.

```
  [uobj_saturation_optionVG]
    ### ### vanGenuchten (https://mooseframework.inl.gov/source/userobjects/PorousFlowCapillaryPressureVG.html)
    type = PorousFlowCapillaryPressureVG   
    alpha = 1e-06
    m = 0.6
    log_extension = True
    sat_lr = 0.25
  []

    type = PorousFlowRelativePermeabilityFLAC
    phase = 0
    m = 2
    s_res = 0
    sum_s_res = 0
```

and below is all my Kernels, nothing special, and `time_unit` is year

```
  [fluid_mass_dot]
    type = PorousFlowMassTimeDerivative   
    variable = porepressure_L1
    fluid_component = 0
    multiply_by_density = true
  []

   [fluid_flux_fullupwind]
     type = PorousFlowAdvectiveFlux   
     variable = porepressure_L1
     gravity = '0 0 -9.81'            
     fluid_component = 0
   []
```

However, when I set excav_perm really high, for example 1e3 higher than the model's avg perm, it converges much better but the drainage/phreatic surface is too much/wider than I wanted.

Everything here should be very straightforward, the only uncertain factor for me right now is excavted_perm, should it be higher or extremely low when the removal subdomain for the excavated block isnt an option?

I tagged you guys since you guys are an expert on this topic: @cpgr @WilkAndy 

Thank you,
Traiwit

![image](https://github.com/idaholab/moose/assets/79562439/75a05fc7-37e2-44d9-ba61-43a6fe8af8dc)

![image](https://github.com/idaholab/moose/assets/79562439/bde9e143-581e-487d-a5ac-b29692799d8a)

**Comment by Traiwit** [comment]
I tried perm_excavated = 0, it actually works BUT only steady-state where `PorousFlowMassTimeDerivative` isnt there

is there a way to make this works with `PorousFlowMassTimeDerivative` on?

**Reply by cpgr** [reply]
Do you do anything to the porosity in the excavated area? Is it set to a small value too? It shows up in the time derivative term during the mass balance.

**Reply by Traiwit** [reply]
Hi, I set poro_excav is 1, poro_rock is usually around 0.05 and poro_cave is 0.25

```
 [mat_porosity]
   type = PorousFlowPorosityFromCSV
   read_prop_user_object = uobj_read_poro_fromCsv
   state = prop_state
   state_values = '1 2' #1 is excav 2 is cave
   poro_state_values = '1 0.25'
 []
```

should this be 0? there is water trapped in that area too (below image is saturation and pwp)

![image](https://github.com/idaholab/moose/assets/79562439/0a534fb8-82c5-452c-b0c4-7d2016eb4baa)

![image](https://github.com/idaholab/moose/assets/79562439/bce0f21e-108d-4550-81a7-7ecd6fe1a904)
